evaluate_prediction.py

```diff
@@ -1,34 +1,30 @@
-import os
-import csv
+#!/usr/bin/env python
+
 import re
 import pandas as pd
 from tqdm import tqdm
 import seaborn as sns
 import matplotlib.pyplot as plt
 from glob import glob
-import numpy as np
 from collections import Counter
 
 import config
 
-def evaluate_tec(original, generated):
+def evaluate_cs(original, generated):
     '''  
     given a list of original and generated events, calculate
-    precision and recall based on translation equivalence.
+    precision and recall of the cardinality score.
     '''
     translation_vectors = []
-    generated_vec = np.array([(
-        float(s['onset']), 
-        int(s['pitch'])
-        ) for s in generated])
-    original_list = [(float(s['onset']), int(s['pitch'])) for s in original]
-    for i in original_list:
-        vectors = generated_vec - i
+    generated_vec = generated[['onset', 'pitch']].values
+    original_list = original[['onset', 'pitch']].values.tolist()
+    for point in original_list:
+        vectors = generated_vec - point
         translation_vectors.extend([tuple(v) for v in vectors])
-    grouped_vectors = dict(Counter(translation_vectors))
-    max_translation = max([grouped_vectors[k] for k in grouped_vectors])
-    recall = (max_translation - 1) / float(len(original) - 1)
-    precision = (max_translation - 1) / float(len(generated) - 1)
+    vector_counts = Counter(translation_vectors)
+    most_common_vector, count = vector_counts.most_common(1)[0]
```

Comment:
Duh, good to learn that the Counter has a most_common function. Much easier! 👍

Reply:
Excellent. Also, I know we don't use `most_common_vector`, but writing it just means people don't need to look up the method 👍